posthog/models/user.py

```diff
@@ -237,6 +228,8 @@ def join(
                 # We don't need to check for ExplicitTeamMembership as none can exist for a completely new member
                 self.current_team = organization.teams.order_by("id").filter(access_control=False).first()
             self.save()
+        if level == OrganizationMembership.Level.OWNER and not self.current_organization.customer_id:
```

Comment:
Seems like we should fetch the relevant org details here, not depend on `current_organization`. If someone is making this change for the user, but the user is currently using a different org, then this would have undesired impacts.

Reply:
This self.current_organization parameter is updated a few lines above to the organization that's passed in as a calling parameter to the function. Is that sufficient? Or is there another way to get the relevant organization information?

Reply:
ah, make sense then!